react-app/src/components/ValidatorDetailScreen/ValidatorDetailScreenAPI.tsx

```diff
@@ -0,0 +1,103 @@
+import { useCallback, useState } from "react";
+import {
+  RequestState,
+  RequestStateError,
+  RequestStateInitial,
+  RequestStateLoaded,
+  RequestStateLoading,
+} from "../../models/RequestState";
+import { useStakingAPI } from "../../api/stakingAPI";
+import { ConnectionStatus, useWallet } from "../../providers/WalletProvider";
+import { useDistributionAPI } from "../../api/distributionAPI";
+import { BigNumberCoin } from "../../models/coin";
+import { useSlashingAPI } from "../../api/slashingAPI";
+import { Validator as ValidatorCosmos } from "../../models/cosmos/staking";
+import { Validator, YourStake } from "./ValidatorDetailScreenModel";
+
+interface ValidatorQuery {
+  (): {
+    fetch: (validatorAddress: string) => Promise<void>;
+    requestState: RequestState<Validator>;
+  };
+}
+
+export const useValidatorQuery: ValidatorQuery = () => {
+  const [requestState, setRequestState] =
+    useState<RequestState<Validator>>(RequestStateInitial);
+
+  const wallet = useWallet();
+  const stakingAPI = useStakingAPI();
+  const slashingAPI = useSlashingAPI();
+  const distributionAPI = useDistributionAPI();
+
+  const fetch = useCallback(
+    // eslint-disable-next-line complexity
+    async (validatorAddress: string) => {
+      setRequestState(RequestStateLoading);
+      try {
+        const promises: Promise<
+          // eslint-disable-next-line @typescript-eslint/no-redundant-type-constituents
+          | ValidatorCosmos
+          | number
+          | Omit<YourStake, "reward">
+          | BigNumberCoin
+          | null
+        >[] = [
+          stakingAPI.getValidator(validatorAddress),
+          distributionAPI.getValidatorExpectedReturn(validatorAddress),
+        ];
+
+        if (wallet.status === ConnectionStatus.Connected) {
+          promises.push(
+            stakingAPI.getDelegation(wallet.account.address, validatorAddress)
+          );
+          promises.push(
+            distributionAPI.getDelegationRewardsByValidator(
+              wallet.account.address,
+              validatorAddress
+            )
+          );
+        }
+
+        const [validatorUntyped, expectedReturn, ...restData] =
```

Comment:
Prevent using spreaded variables as it would be confusing, especially when they are all in different types